ui/src/index.ts

```diff
@@ -93,6 +99,7 @@ window.addEventListener("load", async function() {
       .get();
     const employee = employeeSnapshot.data();
     if (!employee) {
+      userEmail.innerText = userId;
```

Comment:
The logic for the new feature is scattered in two places: here and in `errorOut`. In technical terms, the feature code lacks cohesion. The code for one feature should be all be in the same place if possible. That's because pieces of code for the same feature will tend to change together. If they are not in the same place, mistakes will happen.

In addition, the two parts communicate through a unstructured string: the error message. This is fragile because the message is now part of the contract between the two pieces of code. There's a good chance a developer looking to reword the message will not think to go and update `errorOut` to match. In general, code should not rely on unstructured text written for humans to take decisions.

Let's see what we can do to improve this.

Before the change, `errorOut` had a defined and specific role: display a generic error page. This is great, we don't need to change it. :)

So let's focus here around line 102. It is here that the code has the information that the employee was not found, so it here that the code should act on it. Replacing line 103 with `changePageTo(unknownEmployeePage)` should do the trick.